packages/ontario-design-system-component-library/src/components/ontario-date-input/test/ontario-date-input.spec.tsx

```diff
@@ -120,3 +121,25 @@ describe('ontario-date-input', () => {
 		`);
 	});
 });
+
+describe('date-validation-utils', () => {
+	it('is invalid year value', () => {
+		const isInvalidYearResult = isInvalidYear('two-thousand');
+		expect(isInvalidYearResult).toEqual(true);
+	});
+
+	it('is valid year value', () => {
+		const isValidYear = !isInvalidYear('2000');
+		expect(isValidYear).toEqual(true);
+	});
+
+	it('is valid year when in range', () => {
+		const isValidYear = !isInvalidYear('2000', 1, 9999);
+		expect(isValidYear).toEqual(true);
+	});
+
+	it('is invalid year when out of range', () => {
+		const isValidYear = !isInvalidYear('2000', 1, 1999);
+		expect(isValidYear).toEqual(false);
+	});
+});
```

Comment:
Great test coverage but consider adding a couple edge cases

- Empty strings ('') and null/undefined values.
- Boundary conditions (e.g., exactly at minYear and maxYear).

Reply:
Great idea! 😸